lib/services/northBound/contextServer.js

```diff
@@ -862,12 +862,18 @@ function handleQueryNgsi2(req, res, next) {
 
     logger.debug(context, 'Handling query from [%s]', req.get('host'));
     var contextEntity = {};
-    contextEntity.id = req.query.id;
-    contextEntity.type = req.query.type;
-    var queryAtts = [];
-    if(req.query.attrs) {
-        queryAtts = req.query.attrs.split(',');
+
+    // Heuristic: we are assuming that the entities array in POST /v2/op/query body has only one item
+
+    // FIXME: If the entity has idPattern: '.*' it refers to a whole type (i.e. contextEntity.id is not defined)
```

Comment:
Suggestion in how to improve this part are welcome.

Reply:
Wouldn't be easier just to say that we don't support idPattern so far?

Reply:
Yes, you are right.

This comes from the adaptation of tests like this:

```
    describe('When a context query arrives to the IoT Agent and id query param is not present', function() {
        var options = {
                url: 'http://localhost:' + iotAgentConfig.server.port + '/v2/entities?type=Light',
                method: 'GET',
```

as the equivalent to `GET /v2/entities` without `id=` in `POST /v2/op/query` is using `idPattern`.

But I think is fine to leave this case out of the CB-IOTA integration cases (all the case I know in which CB forwards queries to IOTA are using a particular entity id).

I'll adapt code and tests consequently.

Reply:
Fixed in 5ea1b772657033d0e4691b1b8538e17de5113749